packages/mobile/src/components/core/Switch.tsx

```diff
@@ -0,0 +1,44 @@
+import { useCallback } from 'react'
+
+import { Switch as RNSwitch, SwitchProps as RNSwitchProps } from 'react-native'
+import { useToggle } from 'react-use'
+
+import { light } from 'app/haptics'
+import { useThemeColors } from 'app/utils/theme'
+
+type SwitchProps = RNSwitchProps & {
+  defaultValue?: boolean
+}
+
+export const Switch = (props: SwitchProps) => {
+  const {
+    defaultValue = false,
+    value,
+    onValueChange: onValueChangeProp,
+    ...other
+  } = props
+  const { neutralLight6, neutralLight9, secondary } = useThemeColors()
+  const [isEnabledState, setIsEnabled] = useToggle(defaultValue)
+
+  const isEnabled = value ?? isEnabledState
+
+  const handleValueChange = useCallback(
+    (value: boolean) => {
+      onValueChangeProp?.(value)
+      setIsEnabled(value)
+      light()
```

Comment:
amen. good lookin here. we should add more haptics generally

Reply:
totally agree! esp on the favorite button